fission-web-api/library/Fission/Web/API/User/Challenge/Types.hs

```diff
@@ -0,0 +1,21 @@
+module Fission.Web.API.User.Challenge.Types (Routes (..)) where
+
+import           Fission.Challenge.Types
+
+import           Fission.Web.API.Prelude
+
+import           Fission.User.Username.Types
+import qualified Fission.Web.API.Auth.Types  as Auth
+
+data Routes mode = Routes
+  { recover ::
+      mode
+      :- "recover"
+      :> Summary "Return challenge for account recovery"
+      --
+      :> Capture "Username" Username
+      --
+      :> Auth.HigherOrder
```

Comment:
This line ensures that only fission-registered accounts (and UCANs that are delegated from fission-registered accounts) can make these requests :+1: